enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -81,7 +91,9 @@ const InteractiveDashboardInner = ({
           height={questionHeight}
           plugins={plugins}
           onNavigateBack={onNavigateBackToDashboard}
-        />
+        >
+          {AdHocQuestionView && <AdHocQuestionView />}
```

Comment:
I like option 2) or rather passing function as a child component since users don't have to change the structure to customer their components. By using this approach. We require some markup changes. But if we use option 2) or function as a child component we could keep all the props to `InteractiveQuestion` as is. And only change the way we pass child components.

What do you think?

Reply:
@WiNloSt I'm not sure if I get this correctly, but from the user's perspective they can use this API both ways. They can pass both of these and the component would still work:

```
// Pass in a render function that returns a ReactNode
<InteractiveDashboard renderQuestionView={() => <div />}

// Pass in a React component
<InteractiveDashboard renderQuestionView={QuestionView}
```

Because the render function is a child component. It also would still work if we add in the `({ question: Question })` as the prop.

Reply:
(Another note is that we will allow customizable dashboard layouts soon, so I'd reserve the children for customizable dashboard layouts)

```tsx
<InteractiveDashboard>
  <InteractiveDashboard.Cards />
</InteractiveDashboard>
```

Reply:
Yeah that's what I suggested. I was confused by the mention of option 1 which seems to use `useQuestion` hook, which lifts the state outside of `InteractiveQuestion`. And I couldn't understand how that looks similar to what we're doing here.

I'm re-reading your first comment again, and I think `(props: { question: Question }) => ReactNode` would work great for `InteractiveDashboard`'s `renderQusetionView`

Reply:
@WiNloSt Oh, I just realized you're referring to the proposal for accessing the question's context, and not about this PR. I thought you were talking about the `we could've also rendered this via renderQuestionView` part of the comment, my bad!

> I couldn't understand how that looks similar to what we're doing here.

That proposal is not related directly to this PR, but I referenced it because it proposes exposing a **public-facing** `Question` object. Right now we only have the metabase-lib Question object, which has a lot of internal stuff we don't want to leak. That is why I don't add the `{ question: Question }` yet in this PR - I don't wanna leak that.

PS. Here's [the context of that discussion](https://metaboat.slack.com/archives/C063Q3F1HPF/p1731502645707579?thread_ts=1731485756.247269&cid=C063Q3F1HPF) on the first option.

Reply:
For the "But I'm not quite sure about the API" part, I guess you also refer to the question context proposal and not the PR itself? Let's discuss that in Notion.

Reply:
I was referring to both. Now I'm ok with the InteractiveDashboard API in this PR. For the InteractiveQuestion API let's discuss more on Notion.